src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -17,115 +19,413 @@
 namespace Microsoft.PowerFx
 {
     /// <summary>
-    /// Result of binding an expression. 
+    /// Holds work such as parsing, binding, error checking done on a single expression. 
+    /// Different options require different work. 
+    /// Tracks which work is done so that it is not double repeated.
     /// </summary>
     public class CheckResult : IOperationStatus
     {
-        /// <summary> 
-        /// Return type of the expression. Null if type can't be determined. 
+        #region User Inputs
+        // Captured from user, set once via a Set*() function. 
+
+        /// <summary>
+        /// The source engine this was created against. 
+        /// This is critical for calling back to populate the rest of the results. 
         /// </summary>
-        public FormulaType ReturnType { get; set; }
+        private readonly Engine _engine;
+       
+        // The raw expression test. 
+        private string _expression;
+        
+        private ParserOptions _parserOptions;

```

Comment:
>private ParserOptions _parserOptions; [](http://example.com/codeflow?start=8&length=37)

Do we NEED this input parameter in the results?

Reply:
Yes.  Because some things take just the ParseResults and may want to look back at the options it was parsed against.